src/Silk.Core.Discord/Main.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using DSharpPlus;
+using DSharpPlus.CommandsNext;
+using DSharpPlus.Interactivity.Extensions;
+using MediatR;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Silk.Core.Discord.EventHandlers;
+using Silk.Core.Discord.EventHandlers.MemberAdded;
+using Silk.Core.Discord.EventHandlers.Notifications;
+using Silk.Core.Discord.Types;
+using Silk.Core.Discord.Utilities;
+using Silk.Core.Discord.Utilities.Bot;
+using Silk.Core.Discord.Utilities.HelpFormatter;
+using Silk.Extensions;
+
+namespace Silk.Core.Discord
+{
+    public class Main : IHostedService
+    {
+        public static BotState State { get; private set; } = BotState.Starting;
+        //public static DiscordSlashClient SlashClient { get; } // Soon™ //
+        public static DiscordShardedClient ShardClient { get; private set; }
+        public static string DefaultCommandPrefix { get; } = "s!";
+
+        private static ILogger<Main> _logger;
+        private readonly IServiceProvider _provider;
+
+        public Main(DiscordShardedClient shardClient, ILogger<Main> logger, IServiceProvider provider)
+        {
+            _logger = logger;
+            _provider = provider;
+            ShardClient = shardClient;
+        }
+
+        [MethodImpl(MethodImplOptions.Synchronized)]
+        public static void ChangeState(BotState state)
+        {
+            if (State != state)
+            {
+                _logger.LogDebug("State changed from {State} to {NewState}!", State, state);
+                State = state;
+            }
+        }
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Starting service");
+            await InitializeClientAsync();
+            await InitializeCommandsNextAsync();
+            _logger.LogDebug("Connecting to Discord gateway");
+            await ShardClient.StartAsync();
+            _logger.LogInformation("Connected to Discord gateway as {Username}#{Discriminator}", ShardClient.CurrentUser.Username, ShardClient.CurrentUser.Discriminator);
+        }
+        public async Task StopAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Stopping service");
+            _logger.LogDebug("Disconnecting from Discord gateway");
+            await ShardClient.StopAsync();
+            _logger.LogInformation("Disconnected from Discord gateway");
+        }
+
+        private async Task InitializeClientAsync()
+        {
+            _logger.LogDebug("Initializing client");
+
+            await ShardClient.UseCommandsNextAsync(DiscordConfigurations.CommandsNext);
+            await ShardClient.UseInteractivityAsync(DiscordConfigurations.Interactivity);
```

Comment:
Your function name says that you are initializing your client. However, you are initializing CommandsNext and Interactivity and attaching them to your client. The name is incorrect.